tests/unit/integrations/huggingface/materializers/test_huggingface_tf_model_materializer.py

```diff
@@ -0,0 +1,51 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+from contextlib import ExitStack as does_not_raise
+
+from transformers import TFAutoModelForSequenceClassification, TFPreTrainedModel
+
+from zenml.integrations.huggingface.materializers.huggingface_tf_model_materializer import (
+    HFTFModelMaterializer,
+)
+from zenml.pipelines import pipeline
+from zenml.steps import step
+
+
+def test_huggingface_tf_pretrained_model_materializer(clean_repo):
+    """Tests whether the steps work for the Huggingface Tensorflow Pretrained Model materializer."""
+
+    @step
+    def read_model() -> TFPreTrainedModel:
+        """Reads and materializes a Huggingface Tensorflow Pretrained Model."""
+        return TFAutoModelForSequenceClassification.from_pretrained(
+            "bert-base-cased", num_labels=5
+        )
+
+    @pipeline
+    def test_pipeline(read_model) -> None:
+        """Tests the Huggingface Tensorflow Pretrained Model materializer."""
+        read_model()
+
+    with does_not_raise():
+        test_pipeline(
+            read_model=read_model().with_return_materializers(
+                HFTFModelMaterializer
+            )
+        ).run()
```

Comment:
I see this pattern a lot: 1) define a step with no inputs that returns a specific data type 2) create pipeline with only this step in it 3) make sure the pipeline runs without error. Perhaps this could be generalized.

Reply:
That's the basic version of it, but then there's the extended version which @schustmi encouraged me to do: i.e. actually checking that the artifacts are really functional (via the post-execution checks / calls). Agreed that probably this could all be simplified somehow with an extra function. I'll try to get this in.